src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Workloads/WpfWorkloadDescriptorDataSource.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Concurrent;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Workloads
+{
+    [Export(typeof(IWpfWorkloadDescriptorDataSource))]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class WpfWorkloadDescriptorDataSource : ChainedProjectValueDataSourceBase<ISet<WorkloadDescriptor>>, IWpfWorkloadDescriptorDataSource
```

Comment:
This name is misleading. The type doesn't detect WPF workloads; it detects if the project uses DotNetCoreRazor and either WPF or WinForms. The same goes for the `IWpfWorkloadDescriptorDataSource` interface, the `WpfMissingWorkloadDetector` type, and the fields added to the `MissingSetupComponentRegistrationService`.

Reply:
Perhaps _IWebWorkloadDescriptorDataSource_